src/trust_is_risk.js

```diff
@@ -0,0 +1,282 @@
+    if (changeAmount) {
+      mtx.addOutput(new Output({
+        script: bcoin.script.fromPubkeyhash(bcoin.crypto.hash160(originPubKey)),
+        value: changeAmount
+      }));
+    }
+
+    mtx.addCoin(coin);
+    var success = mtx.scriptVector(coin.script, mtx.inputs[0].script, originKeyRing);
+    assert(success);
+
+    var signedCount = mtx.sign(originKeyRing);
+    assert(signedCount === 1);
+
+    return mtx;
+  }
+
+  // Returns an array of trust-decreasing mutable transaction objects, which reduce a trust
+  // relationship by the amount specified. The payee will receive the amount deducted minus the
+  // transaction fees via P2PKH.
+  // If steal is undefined or set to false, then the `origin` key is expected to be a private key
+  // and the `dest` key is expected to be a public key. If steal is set to true, then `origin` is
+  // expected to be a public key and `dest` is expected to be a private key. The private key will be
+  // used to sign the transaction.
+  getTrustDecreasingMTXs(origin : Key, dest : Key, trustDecreaseAmount : number, payee : ?Entity,
+      steal : ?boolean, fee : ?number) : bcoin$MTX[] {
+    if (steal === undefined) steal = false;
+
+    var signingKeyRing, originKeyRing, destKeyRing;
+    if (!steal) {
+      signingKeyRing = KeyRing.fromPrivate(origin);
+      originKeyRing = KeyRing.fromPrivate(origin);
+      destKeyRing = KeyRing.fromPublic(dest);
+    } else {
+      signingKeyRing = KeyRing.fromPrivate(dest);
+      originKeyRing = KeyRing.fromPublic(origin);
+      destKeyRing = KeyRing.fromPrivate(dest);
+    }
+
+    var originAddress = helpers.pubKeyToEntity(originKeyRing.getPublicKey());
+    var destAddress = helpers.pubKeyToEntity(destKeyRing.getPublicKey());
+
+    if (originAddress === destAddress) throw new Error("Can't decrease self-trust");
+
+    var existingTrustAmount = this.db.getDirectTrustAmount(originAddress, destAddress);
+    if (existingTrustAmount < trustDecreaseAmount) throw new Error("Insufficient trust");
+    
+    var directTrusts = this.db.getSpendableDirectTrusts(originAddress, destAddress);
+    return directTrusts.map((directTrust) => {
+      var decrease = Math.min(trustDecreaseAmount, directTrust.amount);
+      if (decrease === 0) return null;
+      trustDecreaseAmount -= decrease;
+      return this.getTrustDecreasingMTX(directTrust, decrease, payee, signingKeyRing, fee);
+    }).filter(Boolean);
+  }
+
+  getTrustDecreasingMTX(directTrust : DirectTrust, decreaseAmount : number, payee : ?Entity,
+      signingKeyRing : bcoin$KeyRing, fee : ?number) {
+    if (!payee) payee = directTrust.getOriginEntity();
+    if (!fee) fee = 1000; // TODO: estimate this
+
+    var mtx = new MTX({
+      inputs: [
+        Input.fromOutpoint(new Outpoint(directTrust.txHash, directTrust.outputIndex))
+      ],
+      outputs: [new Output({
+        script: bcoin.script.fromPubkeyhash(Address.fromBase58(payee).hash),
+        value: decreaseAmount - fee
```

Comment:
Check that this is positive. If not, make this zero? Throwing an exception may not be appropriate: If multiple txs are spent to achieve the decrease, the last one may be borderline below the fee.